master/internal/rm/kubernetesrm/job_test.go

```diff
@@ -0,0 +1,285 @@
+package kubernetesrm
+
+import (
+	"context"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"gotest.tools/assert"
+	batchV1 "k8s.io/api/batch/v1"
+	k8sV1 "k8s.io/api/core/v1"
+	k8sClient "k8s.io/client-go/kubernetes"
+	typedBatchV1 "k8s.io/client-go/kubernetes/typed/batch/v1"
+	typedV1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/determined-ai/determined/master/internal/config"
+	"github.com/determined-ai/determined/master/internal/rm/rmevents"
+	"github.com/determined-ai/determined/master/internal/sproto"
+	"github.com/determined-ai/determined/master/pkg/cproto"
+	"github.com/determined-ai/determined/master/pkg/device"
+	"github.com/determined-ai/determined/master/pkg/etc"
+	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/tasks"
+)
+
+func createPod(
```

Comment:
nit: move these helper functions to the end of the file

Reply:
sure. i'm really not a fan of the tests in here. they also came from the original pod_test.go. i'll make a ticket to rewrite them.